### ISSUE TITLE
Syntax highlighting unexpectedly

### ISSUE THREAD

**Issue by elo1lson** [lapce/lapce#3461]
## Lapce Version

0.4.2
## System Information

Operating System: Solus OS

## Describe the Bug

https://github.com/user-attachments/assets/6dc47fca-0506-4a55-8a0c-9aec4b60f057

The syntax highlighting in the editor is malfunctioning, rendering all text in white, making it unreadable due to lack of contrast.

## Steps to Reproduce

Open Lapce.
Type any text in the editor.

**Comment by tonydomnickriver** [comment]
i also had that happening with me in windows 10 with rust.

**Comment by elo1lson** [comment]
Did you manage to solve it? I couldn't find a probable cause for this to happen

**Comment by ssuwal** [comment]
It looks like the Semantic Highlighting. I cloned master and just disabled or commented it out and everything works just fine. It is very weird; rust does provide the highlighting from line 0 (which is the first line) while typescript-language-server doesn't.

I disabled `get_semantic_styles` calls from `window_tab.rs` and `doc.rs` for testing. Not sure what the official path would be.

**Comment by elo1lson** [comment]
> ## Lapce Version
> 0.4.2
> 
> ## System Information
> Operating System: Solus OS
> 
> ## Describe the Bug
>  simplescreenrecorder-2024-08-31_23.51.44.mp4 
> The syntax highlighting in the editor is malfunctioning, rendering all text in white, making it unreadable due to lack of contrast.
> 
> ## Steps to Reproduce
> Open Lapce. Type any text in the editor.

**Comment by dzhou121** [comment]
Yeah looks like it's the semantic styles from LSP isn't good enough. Lapce uses semantic styles over the tree-sitter one assuming that LSP should provide a better syntax highlighting.

**Comment by isaiahpettingill** [comment]
Getting a very similar issue in C# on a large solution file on Windows 11. Instead of going white, it just mixes up the colors all over the place and scrambles the type annotations into the main text. I imagine the root issue has something to do with this one.

It highlights correctly until I edit the file

Before:

![Image](https://github.com/user-attachments/assets/30e628bf-5793-4549-a69e-be84d54172df)

After:

![Image](https://github.com/user-attachments/assets/990e1ec6-046a-4720-a267-28579d4e3439)

So a very similar issue.
